juju/main.py

```diff
@@ -73,9 +80,13 @@ def _setup_hooks(charm_dir, charm):
     charm_dir -- A root directory of the charm.
     charm -- An instance of the Charm class.
     """
-    for event_name in charm.on.events().keys():
-        if event_name != 'install':
-            _handle_event_link(charm_dir, event_name)
+    from juju.charm import JujuHookEvent
+
+    def is_hook_event(bound_event):
+        return issubclass(bound_event.event_type, JujuHookEvent)
+
+    for bound_event in filter(is_hook_event, charm.on.events().values()):
```

Comment:
Similar to the comment on the other PR, this seems to make the logic a bit less clear than it could be. I'd go for the straightforward phrasing of this logic, with an if inside the for branch.